airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/poetry/publish/pipeline.py

```diff
@@ -0,0 +1,153 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import configparser
+import io
+import uuid
+from enum import Enum, auto
+from typing import Dict, Optional, Tuple
+
+import tomli
+import tomli_w
+from dagger import Container, Directory
+from pipelines.airbyte_ci.poetry.publish.context import PyPIPublishContext
+from pipelines.consts import PYPROJECT_TOML_FILE_PATH, SETUP_PY_FILE_PATH
+from pipelines.dagger.actions.python.poetry import with_poetry
+from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.steps import Step, StepResult
+
+
+class PackageType(Enum):
+    POETRY = auto()
+    PIP = auto()
+    NONE = auto()
+
+
+class PublishToPyPI(Step):
+    context: PyPIPublishContext
+    title = "Publish package to PyPI"
+
+    def _get_base_container(self) -> Container:
+        return with_poetry(self.context)
+
+    async def _get_package_metadata_from_pyproject_toml(self, dir_to_publish: Directory) -> Optional[Tuple[str, str]]:
```

Comment:
```suggestion
    async def _get_package_metadata_from_pyproject_toml(self, dir_to_publish: Directory) -> Optional[PackageMetadata]:
```

nit: Building a light dataclass with `name` and `version` attribute could boost readability instead of returning a tuple